racket/collects/racket/private/for.rkt

```diff
@@ -722,26 +722,32 @@
   (define (:list-gen l)
     (values car cdr values l pair? #f #f))
 
+  (define (check-mlist l)
+    (unless (or (mpair? l) (null? l)) (raise-argument-error 'in-mlist "(or/c mpair? null?)" l)))
   (define (in-mlist l)
-    (unless (or (null? l) (mpair? l)) (raise-argument-error 'in-mlist "(or/c mpair? null)" l))
+    (check-mlist l)
     (make-do-sequence (lambda () (:mlist-gen l))))
 
   (define (:mlist-gen l)
     (values mcar #f mcdr l mpair? #f #f))
 
-  (define (in-input-port-bytes p)
+  (define (check-in-input-port-bytes p)
```

Comment:
If you will also change Typed Racket anyway, I think it might be a good opportunity to make this a general `check-input-port` that consumes a `who` argument and `p`. The name `'in-input-port-bytes` or `'in-input-port-chars` can then be supplied with the `who` argument.

Reply:
I have some concern about this because I don’t want to duplicate the `who` argument, which then must be supplied both in the expansion and the dynamic sequence constructor.